### PR TITLE
Fix panosinterface bug

### PR DESCRIPTION
**Pull request by AmGeorgiou**
This is a fix for #411.

Setting `enable_dhcp` and `create_dhcp_default_route` to `None` if the module params for these values are not set to `True` does not yield the desired effect. The proposed fix sets them both to `False` unless the module params are set to `True`.

### CODE REVIEW

File: library/panos_interface.py [hunk 1]
```diff
@@ -285,8 +285,8 @@ def main():
         'comment': module.params['comment'],
         'ipv4_mss_adjust': module.params['ipv4_mss_adjust'],
         'ipv6_mss_adjust': module.params['ipv6_mss_adjust'],
-        'enable_dhcp': True if module.params['enable_dhcp'] else None,
-        'create_dhcp_default_route': True if module.params['create_default_route'] else None,
+        'enable_dhcp': True if module.params['enable_dhcp'] else False,
```
Comment (shinmog) [inline library/panos_interface.py]: This change causes a regression preventing `panos_interface` from being able to configure an interface with static IP addresses:

Config:
```yaml
  - name: create static interface eth1/6
    panos_interface:
      provider: '{{ aws_provider }}'
      if_name: 'ethernet1/6'
      enable_dhcp: false
      create_default_route: false
      ip: ['192.168.107.1/24']
      zone_name: 'L3-untrust'
      vr_name: ''
      commit: false
```

Output:
```bash
TASK [create static interface eth1/6] ****************************************************************************************************************
fatal: [10.8.202.114]: FAILED! => {"changed": false, "msg": "Failed apply:  ethernet1/6 -> layer3 -> dhcp-client constraints failed : DHCP interface IP address must be empty\n ethernet1/5 -> layer3 -> dhcp-client is invalid"}
```

Please restore this to `else None`.

File: library/panos_interface.py [hunk 2]
```diff
@@ -285,8 +285,8 @@ def main():
...
+        'create_dhcp_default_route': True if module.params['create_default_route'] else False,
```
Comment (shinmog) [inline library/panos_interface.py]: Similar to the above, you cannot just set this to `False` or it causes the same regression.  The fix that will not cause any regressions is that setting this variable needs to be more explicit and dependent upon what the user specifies for `enable_dhcp` (in pseudo-code):

```
  if it is True:
      then enable it in the spec
  else if enable_dhcp:
      then set it to False in the spec
  else:
      then set it to None in the spec
```

**Comment by AmGeorgiou** [comment]
@shinmog thank you for the speedy response and review.

In trying to update the logic, I got:

`Line break occurred before a binary operator (W503)`

I then tried changing so that the line break occurs after the binary operator and got:

`Line break occurred after a binary operator (W504)`

These rules are contradicting. Is that intentional?

**Comment by AmGeorgiou** [comment]
To validate the logic I'm proposing, I wrote the following script:

```
def test_values(enable_dhcp, create_default_route):
    print "==============="
    print "----Inputs----"
    print "enable_dhcp =", enable_dhcp
    print "create_default_route =", create_default_route
    print "----Outputs----"
    is_dhcp_enabled = (enable_dhcp
                       if isinstance(enable_dhcp, bool)
                       else None)
    print "is_dhcp_enabled=", is_dhcp_enabled
    should_default_route_be_created = (create_default_route
                                       if create_default_route
                                       or (enable_dhcp
                                           and isinstance(create_default_route, bool))
                                       else None)
    print "should_default_route_be_created =", should_default_route_be_created
    print "===============\n\n"


def main():
    possible_combinations = [(True, True), (True, False), (True, None), (False, True), (False, False),
                             (False, None), (None, True), (None, False), (None, None)]
    for combination in possible_combinations:
        test_values(combination[0], combination[1])


if __name__ == '__main__':
    main()
```

Please find the output attached.
[test_values_output.txt](https://github.com/PaloAltoNetworks/ansible-pan/files/3372548/test_values_output.txt)

**Comment by AmGeorgiou** [comment]
Fair play, I've made the changes.

**Pull request merged**